dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -1132,6 +1118,34 @@ public ClassLoader run() {
         });
     }
 
+    @Override
+    @Trivial
+    public boolean hasBeenVisited() {
+        return visited;
+    }
+
+    @Override
+    public Iterator<WebSphereBeanDeploymentArchive> visit() {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
+            Tr.entry(tc, "visit");
+        }
+        visited = true;
+
+        //Runtime Extensions can see everything so they break even our rough dependency graph.
+        //Return no children so they get scanned immediately.
+        if (archive.getType() != ArchiveType.RUNTIME_EXTENSION) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
+                Tr.exit(tc, getHumanReadableName() + " visited, found these children: " + accessibleBDAs);
+            }
+            return accessibleBDAs.iterator();
```

Comment:
a) I don't really like methods with multiple return statements
b) You've got manual exit trace before the return but then the return statement makes another method call